test/specs/core.datasetController.tests.js

```diff
@@ -249,6 +249,69 @@ describe('Chart.DatasetController', function() {
     expect(parsedYValues).toEqual([20, 30]);
   });
 
+  it('should synchronize labels array with X values when chart existing dataset is updated and parsing is off', function() {
```

Comment:
Does this case work properly when parsing is on, or should this fix actually apply in that case too?

Reply:
The bug occurs when parsing is off or has provided keys.

Reply:
@kurkle I totally missed this one but it seems issue occurs with parsing set to `true` as well.
Do you think I should add a 3rd test or maybe change test description to `...when parsing has no provided keys`?

Reply:
After few more tests, it seems I get different results for `parsing: true`. I marked PR as draft until I resolve this.